### ISSUE TITLE
DC/OS Poststart Checks were un-healthy when the UI was up

### ISSUE THREAD

**Issue by orsenthil** [mesosphere/dcos-e2e#740]
I am not sure if this a DC/OS problem or a dcos-e2e problem. At a very high level, I saw the issue consistently with a cluster launched using `dcos-docker` cli, but didn't notice it when I launched using other mechanisms

Here are the steps for reproducing the problem.

1. Download `dcos_generate_config.ee.sh` - It was observed in 1.11.1
2.  dcos-docker create ./dcos_generate_config.ee.sh --license-key ./license.txt --security-mode strict --cluster-id skid3
3. dcos-docker wait --cluster-id skid3
4. dcos-docker web --cluster-id skid3  # launches web and we could see Poststart checks fail.
5. dcos-docker run systemctl list-units --cluster-id skid3 

Has this information that could be interest.

```
dcos-checks-poststart.timer          loaded active     waiting   DC/OS Checks Timer: timer for DC/OS Checks service
dcos-cockroachdb-config-change.timer loaded active     waiting   CockroachDB Set Configuration Timer: Timer to periodically set the CockroachDB configuration
dcos-gen-resolvconf.timer            loaded active     waiting   Generate resolv.conf Timer: periodically updates the network name resolution configuration
dcos-logrotate-master.timer          loaded active     waiting   Logrotate Timer: Timer to trigger every 2 minutes
dcos-signal.timer                    loaded active     waiting   Signal Timer: Timer for DCOS Signal Service
systemd-tmpfiles-clean.timer         loaded active     waiting   Daily Cleanup of Temporary Directories

```

I am not sure if `dcos-docker wait` is waiting for all the units to be up and running.

**Comment by orsenthil** [comment]
Update: The post-checks turned healthy eventually. It took about an hour for my setup.

**Comment by orsenthil** [comment]
https://jira.mesosphere.com/browse/DCOS-20807 - Mesosphere Internal only report _might_ be the bug that is observed here.

**Comment by adamtheturtle** [comment]
I'm not so concerned that the UI is up when before the cluster is healthy.
However, I am concerned that `wait` finishes before the cluster is healthy.

We currently wait for two things on an Enterprise cluster:

1. For `/opt/mesosphere/bin/dcos-diagnostics --diag` to exit with a status code of 0.
2. For DC/OS Test Utils `wait_for_dcos` to return.

**Comment by timaa2k** [comment]
The output mentioned here are not systemd units, they are timers. Unless they trigger they will always be in the `waiting` state.

**Comment by adamtheturtle** [comment]
@timaa2k Does this mean that we should close this issue?

**Comment by timaa2k** [comment]
Yes. If this happens again we would need to output of the `dcos-diagnostics check node-poststart` command or a debug log.

**Comment by orsenthil** [comment]
Thanks @timaa2k  - I could try to reproduce this and share the details of `node-poststart`

**Comment by orsenthil** [comment]
I got to this and found some interesting information.

*dcos-docker run --cluster-id skid3 dcos-diagnostics check node-poststart*

```
{
  "status": 0,
  "checks": {
    "components_master": {
      "output": "",
      "status": 0
    },
    "curl": {
      "output": "",
      "status": 0
    },
    "ip_detect_script": {
      "output": "172.17.0.2 is a valid IPV4 address\n",
      "status": 0
    },
    "journald_dir_permissions": {
      "output": "directory /run/log/journal has the group owner `systemd-journal` and group permissons r-x\n",
      "status": 0
    },
    "mesos_master_replog_synchronized": {
      "output": "",
      "status": 0
    },
    "tar": {
      "output": "",
      "status": 0
    },
    "unzip": {
      "output": "",
      "status": 0
    },
    "xz": {
      "output": "",
      "status": 0
    },
    "zookeeper_serving": {
      "output": "serving\n",
      "status": 0
    }
  }
}

```

*docker exec -it $PUBLIC_AGENT_0 /bin/bash*
*dcos-shell*
*dcos-diagnostics check node-poststart*

```
{
  "status": 3,
  "checks": {
    "components_agent": {
      "output": "",
      "status": 0
    },
    "curl": {
      "output": "",
      "status": 0
    },
    "ip_detect_script": {
      "output": "172.17.0.3 is a valid IPV4 address\n",
      "status": 0
    },
    "journald_dir_permissions": {
      "output": "Error executing systemd journal check: directory /run/log/journal must be in group with Gid 190.\nTry to run: systemd-tmpfiles --create --prefix /run/log/journal\n",
      "status": 3
    },
    "mesos_agent_registered_with_masters": {
      "output": "",
      "status": 0
    },
    "tar": {
      "output": "",
      "status": 0
    },
    "unzip": {
      "output": "",
      "status": 0
    },
    "xz": {
      "output": "",
      "status": 0
    }
  }
}
```

*docker exec -it $AGENT_0 /bin/bash*
*dcos-shell*
*dcos-diagnostics check node-poststart*

```

{
  "status": 3,
  "checks": {
    "components_agent": {
      "output": "",
      "status": 0
    },
    "curl": {
      "output": "",
      "status": 0
    },
    "ip_detect_script": {
      "output": "172.17.0.4 is a valid IPV4 address\n",
      "status": 0
    },
    "journald_dir_permissions": {
      "output": "Error executing systemd journal check: directory /run/log/journal must be in group with Gid 190.\nTry to run: systemd-tmpfiles --create --prefix /run/log/journal\n",
      "status": 3
    },
    "mesos_agent_registered_with_masters": {
      "output": "",
      "status": 0
    },
    "tar": {
      "output": "",
      "status": 0
    },
    "unzip": {
      "output": "",
      "status": 0
    },
    "xz": {
      "output": "",
      "status": 0
    }
  }
}
```

**Comment by orsenthil** [comment]
Here are some related JIRAs - https://jira.mesosphere.com/browse/SOAK-83 
https://jira.mesosphere.com/browse/DCOS_OSS-1361 

Due to a systemd issue https://github.com/systemd/systemd/issues/1397 , /run/log/journal is sometimes owned by root and this is causing a our node-poststart failure.

**Comment by mnaboka** [comment]
unfortunately this seems like a bug in dcos-docker.

Some time ago we noticed that some of our components (dcos-log, dcos-diagnostics) were missing the journald logs. This was actually a customer issue and we also were able to reproduce it in our soak cluster. Turns out there is a sort of race condition in systemd which sometimes results in journald folder being owned by `root:root` instead of `root:systemd-journal`. Our logging services are members of `systemd-journal` group and we assume they can read the journald based on this.

It'd be rather unfair to our customers to change the ownership of system folder without their consent. The journald folder is a part of systemd (DC/OS does not have control over it).

With dcos-checks component we are able to verify the minimum systems requirement for DC/OS. One of which is to have journald folder in the right group `systemd-journal`.

running the command `systemd-tmpfiles --create --prefix /run/log/journal` from dcos-docker as a poststart hook could be a solution. Ideally we should have a docker image with pre-created folder instead of relying on systemd to create it.
